File: src/contracts/system.ts
export type RuntimeKind = "browser-mock" | "tauri";

export interface HealthStatus {
  status: "ok";
  service: "skill-panel-next";
}

export interface SystemAdapter {
  readonly runtime: RuntimeKind;
  healthCheck(): Promise<HealthStatus>;
  schemaVersion(): Promise<number>;
}

export type SystemViewState =
  | {
      status: "ready";
      runtime: RuntimeKind;
      service: string;
      schemaVersion: number;
    }
  | {
      status: "unavailable";
      runtime: RuntimeKind;
      message: string;
    };

File: src-tauri/src/application/system.rs
use crate::infrastructure::database::Database;
use serde::Serialize;

#[derive(Debug, PartialEq, Serialize)]
pub struct HealthStatus {
    pub status: &'static str,
    pub service: &'static str,
}

pub fn health_check() -> HealthStatus {
    HealthStatus {
        status: "ok",
        service: "skill-panel-next",
    }
}

pub fn schema_version(database: &Database) -> Result<i64, String> {
    database.initialize()
}
